src/claimtrie.cpp

```diff
@@ -1237,20 +1212,80 @@ bool CClaimTrieCache::recursiveComputeMerkleHash(CClaimTrieNode* tnCurrent, std:
     return true;
 }
 
-uint256 CClaimTrieCache::getMerkleHash() const
+bool CClaimTrieCache::recursiveComputeMerkleHashBinaryTree(CClaimTrieNode* tnCurrent, std::string sPos, bool forceCompute) const
+{
+    if (tnCurrent->empty()) {
+        cacheHashes[sPos] = _noChildrenOrClaims;
+        return true;
+    }
+
+    std::vector<uint256> childHashes;
+    for (nodeMapType::iterator it = tnCurrent->children.begin(); it != tnCurrent->children.end(); ++it) {
+        std::stringstream ss;
```

Comment:
stringstream is slowest C++ string manipulation class, better not use it in time expensive functions. You can use string + char which is faster.

Reply:
Sorry, the stringstream was there in the existing method that I copied to start with. There are quite a few of those in the file, unfortunately. I completely agree with you that it's both slow and a horribly asinine way to convert a char to a string. I've been dragging my feet a little on completely refactoring the trie searches (aka, moving them all to a search method that all these other methods call); I've been waiting for the upstream merge to happen before doing some major refactoring on the trie. It would eliminate all stringstream calls.